python/src/uagents/registration.py

```diff
@@ -293,3 +293,42 @@ async def register(
         except Exception as e:
             self._logger.error(f"Failed to register on Almanac contract: {e}")
             raise
+
+
+class AgentStatusUpdate(BaseModel):
+    agent_address: str
+    is_active: bool
+
+    def sign(self, identity: Identity):
+        digest = self._build_digest()
+        self.signature = identity.sign_digest(digest)
+
+    def verify(self) -> bool:
+        if self.signature is None:
+            raise ValueError("Status update signature is missing")
+        return Identity.verify_digest(
+            self.agent_address, self._build_digest(), self.signature
+        )
+
+    def _build_digest(self) -> bytes:
+
+        sha256 = hashlib.sha256()
+        sha256.update(
+            json.dumps(
+                self.model_dump(exclude={"signature"}),
+                sort_keys=True,
+                separators=(",", ":"),
+            ).encode("utf-8")
+        )
+        return sha256.digest()
+
+
+async def update_agent_status(update: AgentStatusUpdate, almanac_api: str):
+    async with aiohttp.ClientSession() as session:  # noqa: SIM117
+        async with session.post(
```

Comment:
It may be worth adding retries here

Reply:
It depends on how important this information is. We need to accept the fact that that there will always be situations in which an agent cannot send an update to the Almanac API but I am still for adding retries here. I'd say 3 should be enough with the exponential backoff.